source/game_sa/Entity/Vehicle/Vehicle.cpp

```diff
@@ -3267,10 +3318,16 @@ void CVehicle::ProcessBikeWheel(CVector& wheelFwd, CVector& wheelRight, CVector&
         wheelStatus);
 }
 
-// return nearest wheel?
 // 0x6D7BC0
-int32 CVehicle::FindTyreNearestPoint(CVector2D point) {
-    return ((int32(__thiscall*)(CVehicle*, float, float))0x6D7BC0)(this, point.x, point.y);
+eCarWheel CVehicle::FindTyreNearestPoint(CVector2D point) {
+    const auto relativePt = point - GetPosition2D();
+    const bool isRight = relativePt.Dot(GetForward()) <= 0.f; // TODO: This doesn't make a lot of sense, why is Y used for left/right?
+    if (IsBike()) {
+        return isRight ? CAR_WHEEL_FRONT_RIGHT : CAR_WHEEL_FRONT_LEFT;
+    }
+    const bool isFront = relativePt.Dot(GetRight()) <= 0.f; // TODO: Same here, why is X used for front/rear?
+    return isRight ? isFront ? CAR_WHEEL_FRONT_RIGHT : CAR_WHEEL_REAR_RIGHT
+        : isFront ? CAR_WHEEL_REAR_LEFT : CAR_WHEEL_FRONT_LEFT;
```

Comment:
Why is the formatting broken?
```suggestion

    return isRight
        ? isFront ? CAR_WHEEL_FRONT_RIGHT : CAR_WHEEL_REAR_RIGHT
        : isFront ? CAR_WHEEL_REAR_LEFT : CAR_WHEEL_FRONT_LEFT;
```